document-generator-prosecution/src/main/java/uk/gov/companieshouse/document/generator/prosecution/service/ProsecutionService.java

```diff
@@ -0,0 +1,123 @@
+package uk.gov.companieshouse.document.generator.prosecution.service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import uk.gov.companieshouse.api.InternalApiClient;
+import uk.gov.companieshouse.api.error.ApiErrorResponseException;
+import uk.gov.companieshouse.api.handler.exception.URIValidationException;
+import uk.gov.companieshouse.api.model.ApiResponse;
+import uk.gov.companieshouse.api.model.prosecution.defendant.DefendantApi;
+import uk.gov.companieshouse.api.model.prosecution.offence.OffenceApi;
+import uk.gov.companieshouse.api.model.prosecution.prosecutioncase.ProsecutionCaseApi;
+import uk.gov.companieshouse.document.generator.prosecution.exception.ProsecutionServiceException;
+import uk.gov.companieshouse.document.generator.prosecution.mapping.mappers.ApiToDefendantMapper;
+import uk.gov.companieshouse.document.generator.prosecution.mapping.mappers.ApiToOffenceMapper;
+import uk.gov.companieshouse.document.generator.prosecution.mapping.mappers.ApiToProsecutionCaseMapper;
+import uk.gov.companieshouse.document.generator.prosecution.mapping.model.defendant.Defendant;
+import uk.gov.companieshouse.document.generator.prosecution.mapping.model.offence.Offence;
+import uk.gov.companieshouse.document.generator.prosecution.mapping.model.prosecutioncase.ProsecutionCase;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
+
+import static uk.gov.companieshouse.document.generator.prosecution.ProsecutionDocumentInfoService.MODULE_NAME_SPACE;
+
+@Service
+public class ProsecutionService {
```

Comment:
No unit test coverage for this service

Reply:
Unit tests have been added for this class.